root/static/scripts/common/components/Relationship.js

```diff
@@ -68,15 +68,17 @@ const RelationshipContent = ({
   const linkType = linkedEntities.link_type[relationship.linkTypeID];
   let entity0 = relationship.entity0;
   let entity1 = relationship.entity1;
+  const type0 = relationship.source_type || linkType.type0;
+  const type1 = relationship.target_type || linkType.type1;
```

Comment:
I see in `Edit::Relationship::Create` `source_type` and `target_type` are hard coded to `type0` & `type1` respectively, but in `Entity::Relationship` it depends on the direction of the relationship. So I'd think this should be something like
```suggestion
  const type0 = linkType.type0 ||
    (direction === 'backward' ? relationship.target_type : relationship.source_type); 
  const type1 = linkType.type1 ||
    (direction === 'backward' ? relationship.source_type : relationship.target_type); 
```